_posts/2018-11-02-cpp-array.md

```diff
@@ -0,0 +1,153 @@
+---
+title: C++ persistent containers - array
+author: igchor
+layout: post
+identifier: cpp_array
+---
+
+### Introduction
+
+Until now, our C++ bindings were missing one important component - persistent
+containers. In 1.5 release we have introduced the first one - [pmem::obj::array][cpp_array].
+This container is currently placed in `experimental` namespace and folder. It has
+almost the same functionality as `std::array` from C++11 but takes care of adding
+elements to a transaction and guarantees a fixed layout. API for `pmem::obj::array`
+and `std::array` is the same, except for the following:
+- `pmem::obj::array` defines [range()][cpp_array_range] method (will be described below)
+- `pmem::obj::array` does not mark any non-const function as `noexcept` -
+	elements must be added to a transaction which could result in an exception
+
+If you want to store sequence (which length is known at compile time) of objects
+on persistent memory you should always use `pmem::obj::array`. When using simple
```

Comment:
In contrast to simple (maybe plain?) arrays, `pmem::obj:array` automatically adds modified elements to the enclosing transaction.